src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -301,17 +324,27 @@ private static (RoutePattern resolvedRoutePattern, IDictionary<string, string?>
         return (attributeRoutePattern, resolvedRequiredValues ?? action.RouteValues);
     }
 
-    private static void AddActionDataToBuilder(
+    private void AddActionDataToBuilder(
         EndpointBuilder builder,
         HashSet<string> routeNames,
         ActionDescriptor action,
         string? routeName,
         RouteValueDictionary? dataTokens,
         bool suppressLinkGeneration,
         bool suppressPathMatching,
+        IReadOnlyList<Action<EndpointBuilder>> groupConventions,
         IReadOnlyList<Action<EndpointBuilder>> conventions,
         IReadOnlyList<Action<EndpointBuilder>> perRouteConventions)
     {
+        // REVIEW: The RouteEndpointDataSource adds HttpMethodMetadata before running group conventions
```

Comment:
I think it's goodness to give access to this metadata to group conventions. I don't anticipate that it'll be as required here as it is in REDS, where it's needed for OpenAPI support. IMO, nice to have but not groundbreaking.

Reply:
> I think it's goodness to give access to this metadata to group conventions. I don't anticipate that it'll be as required here as it is in REDS, where it's needed for OpenAPI support. IMO, nice to have but not groundbreaking.

Open API fixes it with Finally 😄

Reply:
We should definitely fix this inconsistency. RouteEndpointDataSource definitely has the nicer, more convenient behavior. Maybe we can sneak this into #43225.

Reply:
@halter73 Sure -- I can adjust. Do any other conventions rely on this particular metadata? OpenAPI was the only one that came to mind.

Reply:
I think we’d need a scenario where it matters. This code needs to be re-worked to do that I think. We don’t want group conventions to override endpoint metadata in attributes and metadata set from the action descriptor

Reply:
> This code needs to be re-worked to do that I think. We don’t want group conventions to override endpoint metadata in attributes and metadata set from the action descriptor

I agree with this part.

> I think we’d need a scenario where it matters. 

I think consistency is enough of a reason to say we ought to do this. I can agree that fixing this isn't the highest priority thing, and if it's too complicated it might not be worth doing now.

Reply:
And yes, WithOpenApi is the only thing I know of that already depends on this and it *is* fixed by `Finally`.